doc/GenAIUsageGuidelines.md

```diff
@@ -0,0 +1,128 @@
+-->
+
+# Eclipse OMR Generative Artificial Intelligence Usage Guidelines
+
+The Eclipse OMR project acknowledges that some contributions may be authored in
+whole or in part from generative artificial intelligence (GenAI) coding
+assistants.  Such contributions are permitted per the
+[Eclipse Foundation Generative AI Usage Guidelines](https://www.eclipse.org/projects/guidelines/genai/)
+which describes contributor and committer responsibilities, copyright and
+licensing considerations, and attribution requirements for GenAI assisted
+contributions.
+
+All contributors are encouraged to read and understand the advice and
+guidelines therein before submitting a contribution to Eclipse OMR even
+partially generated by an AI coding assistant.
+
+The Eclipse OMR project is subject to those policies and provides additional
+guidelines described below for contributors and committers governing
+GenAI-assisted contributions.
+
+## Attribution
+
+Contributions that include GenAI-assisted content must disclose that authorship
+in each file where GenAI content is present.  Contributors must include a
+statement like the following just below the copyright header at the beginning
+of the file that acknowledges their AI assistant (assuming such a statement
+does not already exist).
+
+```
+// Some sections generated by <YOUR AI ASSISTANT>
+```
+
+In addition, individual commits that include GenAI authored content must
+attribute that at the end of the commit message (before any human authorship
+signatures).
+
+```
+Co-authored by <YOUR AI ASSISTANT>
+```
+
+## Tagging
+
+Some contributors’ employers may require their AI-generated code and
+documentation contributions to be annotated with inline tags providing
+additional metadata about the generated content for legal and license
+attribution reasons. Eclipse OMR will accept contributions that include such
+metadata with some guidance below for contributors and committers.
+
+### Tag Usage
+
+Inline tagging of code or documentation generated by coding assistants is not
+mandatory.
+
+If tags are used, they must clearly demarcate the boundaries of GenAI-assisted
+content (e.g., via paired "begin" and "end" code comments). These boundaries
+may span consecutive lines, functions, paragraphs, or entire files. The term
+“assisted” is an important qualifier to describe the enclosed content because
+it may contain a blend of AI-generated content and human contributions. The
+“assisted” distinction is necessary because it may not be possible to clearly
+identify just the AI-generated content in the contribution, and it allows for
+human changes to the demarcated contribution over time while preserving the
+meaning of the tags.
+
+### Tag Format
+
+At this time, Eclipse OMR does not specify the format of the tags, except that
+they must not contain any corporate-specific information or other identifiers
+that have no meaning beyond the original contributor.  Information about the
```

Comment:
> they must not contain any corporate-specific information or other identifiers
that have no meaning beyond the original contributor.

It's not clear to me what this means.  If my employer, HAL Corp., say, requires me to tag code generated by its AI assistant as `// {begin|end} generated-by-hal-ai` or `// {begin|end} hal-generated-content-is-the-best`, is that corporate-specific information?  Is the idea that the tagging should indicate that the demarcated region was generated by some AI, but that it must not contain a trademark of any sort?

Reply:
My reading of that section is that the tags are free to refer the specific AI assistant which is likely to contain or imply the vendor of that assistant, HAL in your example, which is fine whether you work for HAL or simply use HAL's assistant.

I think the prohibition is against things that have no meaning to anyone who isn't privy to things behind the corporate-specific tag (e.g. references to issues or other proprietary information accessible only to employees of a certain corporation). Also inappropriate would be any subjective commentary like "is-the-best" in your example.

Reply:
That makes sense.  If that's what was intended, I think it would be worth clarifying.

Reply:
Keith's interpretation is correct.  I (hopefully) added some clarification to the text in the forced push.